server/src/main/java/io/spine/server/entity/storage/EntityRecordWithColumns.java

```diff
@@ -21,188 +21,141 @@
-        Columns columns = recordStorage.columns();
-        Map<ColumnName, @Nullable Object> storageFields = columns.valuesIn(entity);
-        return of(record, storageFields);
-    }
-
-    /**
-     * Wraps a passed entity record.
+     * Creates the new instance of {@code EntityRecordWithColumns} by evaluating the values
+     * of the passed columns for the passed entity.
      *
-     * <p>Such instance of {@code EntityRecordWithColumns} will contain no storage fields.
-     */
-    public static EntityRecordWithColumns of(EntityRecord record) {
-        return new EntityRecordWithColumns(record, Collections.emptyMap());
-    }
-
-    /**
-     * Creates a new instance from the passed record and storage fields.
-     */
-    @VisibleForTesting
-    public static EntityRecordWithColumns
-    of(EntityRecord record, Map<ColumnName, Object> storageFields) {
-        return new EntityRecordWithColumns(record, storageFields);
-    }
-
-    /**
-     * Returns the enclosed entity record.
+     * @param entity
+     *         the entity to use as a provider of the record identifier and the column values
+     * @param record
+     *         the record prepared for storage
+     * @param <I>
+     *         the type of the entity identifiers
+     * @param <E>
+     *         the type of the entity
+     * @return a new instance of {@code EntityRecordWithColumns}
      */
-    public EntityRecord record() {
-        return record;
+    public static <I, S extends EntityState<I>, E extends Entity<I, S>> EntityRecordWithColumns<I>
+    create(E entity, EntityRecord record) {
+        checkNotNull(entity);
+        checkNotNull(record);
+        EntityRecordSpec<I, S, E> recordSpec = EntityRecordSpec.of(entity);
+        Map<ColumnName, @Nullable Object> storageFields = recordSpec.valuesIn(entity);
+        return new EntityRecordWithColumns<>(entity.id(), record, storageFields);
     }
 
     /**
-     * Obtains the names of storage fields in the record.
+     * Creates the new instance of {@code EntityRecordWithColumns} using the pre-created
+     * entity record and the entity identifier.
      *
-     * @return the storage field names
+     * <p>This method considers only the values of the
+     * {@linkplain EntityRecordColumn lifecycle columns}.
+     *
+     * @param id
+     *         the identifier of the entity
+     * @param record
+     *         the record to store; it is also used as a source for the lifecycle column values
+     * @param <I>
+     *         the type of the identiiers
+     * @return a new instance of {@code EntityRecordWithColumns}
      */
-    public ImmutableSet<ColumnName> columnNames() {
-        return ImmutableSet.copyOf(storageFields.keySet());
+    public static <I> EntityRecordWithColumns<I> create(I id, EntityRecord record) {
+        checkNotNull(id);
+        checkNotNull(record);
+        LifecycleFlags flags = record.getLifecycleFlags();
+        ImmutableMap<ColumnName, Object> lifecycleValues =
```

Comment:
In relation to this fragment, please see my earlier comment in `EntityStorageRecord` regarding the chances of consolidating the knowledge about lifecycle columns under a class.